NFIQ2/NFIQ2Algorithm/include/nfiq2/modelinfo.hpp

```diff
@@ -0,0 +1,114 @@
+#ifndef NFIQ2_MODELINFO_H_
+#define NFIQ2_MODELINFO_H_
+
+#include <string>
+#include <unordered_map>
+
+namespace NFIQ {
+/** Model Class containing Model Information */
+class ModelInfo {
+    public:
+	ModelInfo(const std::string &modelInfoFilePath);
+	~ModelInfo();
+
+	/**
+	 * @brief
+	 * Getter for private member modelName
+	 *
+	 * @return
+	 * Returns model name
+	 */
+	std::string getModelName() const;
+
+	/**
+	 * @brief
+	 * Getter for private member modelTrainer
+	 *
+	 * @return
+	 * Returns model trainer
+	 */
+	std::string getModelTrainer() const;
+
+	/**
+	 * @brief
+	 * Getter for private member modelDescription
+	 *
+	 * @return
+	 * Returns model description
+	 */
+	std::string getModelDescription() const;
+
+	/**
+	 * @brief
+	 * Getter for private member modelVersion
+	 *
+	 * @return
+	 * Returns model version
+	 */
+	std::string getModelVersion() const;
+
+	/**
+	 * @brief
+	 * Getter for private member modelPath
+	 *
+	 * @return
+	 * Returns model path
+	 */
+	std::string getModelPath() const;
+
+	/**
+	 * @brief
+	 * Getter for private member modelHash
+	 *
+	 * @return
+	 * Returns model hash
+	 */
+	std::string getModelHash() const;
```

Comment:
Consider making the members public or this a struct so you can eliminate these methods and just pick them off. Later comment will ask to make members const. Not terrible, but saves lots of typing.